
These are the changes made to the openssl.h that Configure creates.

--- opensslconf.h-orig-m	Sat Mar 28 00:36:31 2015
+++ opensslconf.h	Sat Mar 28 14:42:46 2015
@@ -1,6 +1,11 @@
 /* opensslconf.h */
 /* WARNING: Generated automatically from opensslconf.h.in by Configure. */
 
+#ifdef	OPENSSL_SUNW_PREFIX
+/* Rename all external symbols */
+#include <openssl/sunw_prefix.h>
+#endif
+
 #ifdef  __cplusplus
 extern "C" {
 #endif
@@ -101,9 +106,12 @@
 
 #endif /* OPENSSL_DOING_MAKEDEPEND */
 
+#ifndef	_BOOT
 #ifndef OPENSSL_THREADS
 # define OPENSSL_THREADS
 #endif
+#endif
+
 #ifndef OPENSSL_NO_STATIC_ENGINE
 # define OPENSSL_NO_STATIC_ENGINE
 #endif
@@ -214,7 +222,11 @@
 
 #if !(defined(VMS) || defined(__VMS)) /* VMS uses logical names instead */
 #if defined(HEADER_CRYPTLIB_H) && !defined(OPENSSLDIR)
+#if defined(__sparcv9) || defined(__x86_64)
+#define ENGINESDIR "/lib/openssl/engines/64"
+#else
 #define ENGINESDIR "/lib/openssl/engines"
+#endif
 #define OPENSSLDIR "/etc/openssl"
 #endif
 #endif
@@ -261,21 +273,39 @@
 /* If this is set to 'unsigned int' on a DEC Alpha, this gives about a
  * %20 speed up (longs are 8 bytes, int's are 4). */
 #ifndef DES_LONG
+#if defined(__sparcv9) || defined(__x86_64)
+#define DES_LONG unsigned int
+#else
 #define DES_LONG unsigned long
 #endif
 #endif
+#endif
 
 #if defined(HEADER_BN_H) && !defined(CONFIG_HEADER_BN_H)
 #define CONFIG_HEADER_BN_H
+/*
+ * OpenSSL revision 1.521 from 2005-12-15 in OpenSSL_1_0_0-stable branch changed
+ * 64 bit sparcv9 configuration from SIXTY_FOUR_BIT_LONG to BN_LLONG.
+ */
+#if defined(__x86_64) || defined(_BOOT)
+#undef BN_LLONG
+#else
 #define BN_LLONG
+#endif
 
 /* Should we define BN_DIV2W here? */
 
 /* Only one for the following should be defined */
+#if defined(__x86_64)
+#define SIXTY_FOUR_BIT_LONG
+#undef THIRTY_TWO_BIT
+#else
 #undef SIXTY_FOUR_BIT_LONG
 #undef SIXTY_FOUR_BIT
 #define THIRTY_TWO_BIT
 #endif
+#undef SIXTY_FOUR_BIT
+#endif
 
 #if defined(HEADER_RC4_LOCL_H) && !defined(CONFIG_HEADER_RC4_LOCL_H)
 #define CONFIG_HEADER_RC4_LOCL_H
